### DISCUSSION TITLE
Server keeps freezing after backup.

### DISCUSSION THREAD

**Discussion by lyk4ndna** [mbround18/valheim-docker#531] [Help!] [answered]
For some reason, after every back up it keeps backing up multiple times and the server freezes. I have to restart the computer and then re-up docker to get it to go again, until the next morning where it tries to back-up again.

Below is the instance of the backup: 

> valheim_1  | 
valheim_1  | ==> /home/steam/valheim/logs/auto-backup.out <==
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220219-081501-backup.tar.gz
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220219-083001-backup.tar.gz
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220219-084501-backup.tar.gz
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220219-090001-backup.tar.gz
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220219-091501-backup.tar.gz
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220219-093001-backup.tar.gz
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220219-094501-backup.tar.gz
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220219-100001-backup.tar.gz
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220219-101501-backup.tar.gz
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220219-103001-backup.tar.gz
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220219-104501-backup.tar.gz
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220219-110001-backup.tar.gz
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220219-111501-backup.tar.gz
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220219-113001-backup.tar.gz
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220219-114501-backup.tar.gz
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220219-120001-backup.tar.gz
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220219-121501-backup.tar.gz
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220219-123001-backup.tar.gz
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220219-124501-backup.tar.gz
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220219-130001-backup.tar.gz
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220219-131501-backup.tar.gz
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220219-133001-backup.tar.gz
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220219-134501-backup.tar.gz
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220219-140001-backup.tar.gz
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220219-141501-backup.tar.gz
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220219-143001-backup.tar.gz
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220219-144501-backup.tar.gz
^CGracefully stopping... (press Ctrl+C again to force)
Stopping valheim_valheim_1   ... done

Here is the docker file: 

> version: "3"
services:
  valheim:
    image: mbround18/valheim:latest
    ports:
      - 2456:2456/udp
      - 2457:2457/udp
      - 2458:2458/udp
    environment:
      - PORT=2456
      - NAME="WORLD"
      - WORLD="WORLD"
      - PASSWORD="PASSWORD"
      - TYPE=ValheimPlus
      - "MODS=
---MODS---
       "
      - TZ=America/Chicago
      - PUBLIC=1
      - AUTO_UPDATE=1
      - AUTO_UPDATE_SCHEDULE="5 8 * * *"
      - AUTO_BACKUP=1
      - AUTO_BACKUP_SCHEDULE="*/15 * * * *"
      - AUTO_BACKUP_DAYS_TO_LIVE=1
      - AUTO_BACKUP_REMOVE_OLD=1
      - AUTO_BACKUP_PAUSE_WITH_NO_PLAYERS=0
      - WEBHOOK_URL="URL"
    volumes:
    - ./valheim/saves:/home/steam/.config/unity3d/IronGate/Valheim
    - ./valheim/server:/home/steam/valheim
    - ./valheim/backups:/home/steam/backups

**Comment by mbround18** [comment, marked as answer]
huh @lyk4ndna the backup process is completely different from the server running process. When it freezes, is your cpu spiked? ram fully utilized? or out of disk space?

**Reply by lyk4ndna** [reply]
It could be RAM, because it is almost maxed out or it spikes when the backup is made. It is also making a backup when there is no players in the server. Does the pause backup if no players need to be 0 or 1?

**Reply by mbround18** [reply]
@lyk4ndna set `AUTO_BACKUP_PAUSE_WITH_NO_PLAYERS=1` to pause with no players.

**Reply by lyk4ndna** [reply]
@mbround18 I changed it. I definitely read it wrong, but this docker file has been configured like this for almost 6 months. 

Only recently has it started freezing up on the back-up and stuck in that loop. It seems to happen after it checks for an update in the morning. Then it goes into that backup loop. After I stop and start it again, it will run, we can hop on, and it’ll be fine. 

Attaching a photo of the system monitor after the docker file first initializing.
![E5C55583-2972-4949-BCAB-52D58A39CC59](https://user-images.githubusercontent.com/89585113/155251178-1efe17d1-68a1-41b1-90c2-b870ceb0cf31.jpeg)

It is running on a rockpi x (4GB RAM, 64 GB EMMC).

**Reply by mbround18** [reply]
Hmm I wonder if a dependency changed or something, give me a bit to investigate

**Reply by lyk4ndna** [reply]
@mbround18  I changed the piece to auto pause if there is no players and it seems to skip the backup, but then it will do the same thing again where it will go into a backup loop. 

It seems something happened around 00:41 on the SteamnetworkingSockets, then right after that it would kick off backups.

<details>
<summary>Output (Click to open)</summary>

valheim_1  | 03/03/2022 00:13:39: Game server disconnected
valheim_1  |  
valheim_1  | (Filename: ./Runtime/Export/Debug/Debug.bindings.h Line: 39)
valheim_1  | 
valheim_1  | 03/03/2022 00:13:43: Game server connected
valheim_1  |  
valheim_1  | (Filename: ./Runtime/Export/Debug/Debug.bindings.h Line: 39)
valheim_1  | 
valheim_1  | 
valheim_1  | ==> /home/steam/valheim/logs/auto-backup.out <==
valheim_1  | [Valheim][steam]: Skipping backup, no players are online.
valheim_1  | 
valheim_1  | ==> /home/steam/valheim/logs/valheim_server.log <==
valheim_1  | 03/03/2022 00:18:59:  Connections 0 ZDOS:1120684  sent:0 recv:0
valheim_1  |  
valheim_1  | (Filename: ./Runtime/Export/Debug/Debug.bindings.h Line: 39)
valheim_1  | 
valheim_1  | Unloading 0 Unused Serialized files (Serialized files now loaded: 2)
valheim_1  | 
valheim_1  | Unloading 0 unused Assets to reduce memory usage. Loaded Objects now: 1263575.
valheim_1  | Total: 3316.310998 ms (FindLiveObjects: 334.101537 ms CreateObjectMapping: 62.835823 ms MarkObjects: 2916.542202 ms  DeleteObjects: 2.797949 ms)
valheim_1  | 
valheim_1  | 03/03/2022 00:19:12: clone 9992
valheim_1  |  
valheim_1  | (Filename: ./Runtime/Export/Debug/Debug.bindings.h Line: 39)
valheim_1  | 
valheim_1  | 03/03/2022 00:20:04: Saved 1120684 zdos
valheim_1  |  
valheim_1  | (Filename: ./Runtime/Export/Debug/Debug.bindings.h Line: 39)
valheim_1  | 
valheim_1  | 03/03/2022 00:20:05: World saved ( 52020.452ms )
valheim_1  |  
valheim_1  | (Filename: ./Runtime/Export/Debug/Debug.bindings.h Line: 39)
valheim_1  | 
valheim_1  | 03/03/2022 00:29:42:  Connections 0 ZDOS:1120684  sent:0 recv:0
valheim_1  |  
valheim_1  | (Filename: ./Runtime/Export/Debug/Debug.bindings.h Line: 39)
valheim_1  | 
valheim_1  | Unloading 0 Unused Serialized files (Serialized files now loaded: 2)
valheim_1  | 
valheim_1  | ==> /home/steam/valheim/logs/auto-backup.out <==
valheim_1  | [Valheim][steam]: Skipping backup, no players are online.
valheim_1  | 
valheim_1  | ==> /home/steam/valheim/logs/valheim_server.log <==
valheim_1  | 
valheim_1  | Unloading 0 unused Assets to reduce memory usage. Loaded Objects now: 1263575.
valheim_1  | Total: 68361.125268 ms (FindLiveObjects: 45976.490447 ms CreateObjectMapping: 85.348947 ms MarkObjects: 22274.845013 ms  DeleteObjects: 24.399386 ms)
valheim_1  | 
valheim_1  | 03/03/2022 00:39:09: Possible events:0
valheim_1  |  
valheim_1  | (Filename: ./Runtime/Export/Debug/Debug.bindings.h Line: 39)
valheim_1  | 
valheim_1  | 03/03/2022 00:40:51:  Connections 0 ZDOS:1120684  sent:0 recv:0
valheim_1  |  
valheim_1  | (Filename: ./Runtime/Export/Debug/Debug.bindings.h Line: 39)
valheim_1  | 
valheim_1  | Unloading 0 Unused Serialized files (Serialized files now loaded: 2)
valheim_1  | 
valheim_1  | Unloading 0 unused Assets to reduce memory usage. Loaded Objects now: 1263575.
valheim_1  | Total: 4521.610827 ms (FindLiveObjects: 482.047769 ms CreateObjectMapping: 796.048494 ms MarkObjects: 3241.083402 ms  DeleteObjects: 2.394450 ms)
valheim_1  | 
valheim_1  | 03/03/2022 00:41:05: clone 9901
valheim_1  |  
valheim_1  | (Filename: ./Runtime/Export/Debug/Debug.bindings.h Line: 39)
valheim_1  | 
valheim_1  | 
valheim_1  | ==> /home/steam/valheim/logs/valheim_server.err <==
valheim_1  | src/steamnetworkingsockets/clientlib/steamnetworkingsockets_lowlevel.cpp (1852) : Assertion Failed: SteamnetworkingSockets service thread waited 114ms for lock!  This directly adds to network latency!  It could be a bug, but it's usually caused by general performance problem such as thread starvation or a debug output handler taking too long.
valheim_1  | 
valheim_1  | ==> /home/steam/valheim/logs/valheim_server.log <==
valheim_1  | src/steamnetworkingsockets/clientlib/steamnetworkingsockets_lowlevel.cpp (1852) : Assertion Failed: SteamnetworkingSockets service thread waited 114ms for lock!  This directly adds to network latency!  It could be a bug, but it's usually caused by general performance problem such as thread starvation or a debug output handler taking too long.
valheim_1  | 
valheim_1  | ==> /home/steam/valheim/logs/valheim_server.err <==
valheim_1  | src/steamnetworkingsockets/clientlib/steamnetworkingsockets_lowlevel.cpp (1852) : Assertion Failed: SteamnetworkingSockets service thread waited 1948ms for lock!  This directly adds to network latency!  It could be a bug, but it's usually caused by general performance problem such as thread starvation or a debug output handler taking too long.
valheim_1  | 
valheim_1  | ==> /home/steam/valheim/logs/auto-backup.out <==
valheim_1  | parse error: Invalid numeric literal at line 1, column 6
valheim_1  | /home/steam/scripts/auto_backup.sh: line 19: [: : integer expression expected
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220303-004507-backup.tar.gz
valheim_1  | parse error: Invalid numeric literal at line 1, column 6
valheim_1  | /home/steam/scripts/auto_backup.sh: line 19: [: : integer expression expected
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220303-010006-backup.tar.gz
valheim_1  | parse error: Invalid numeric literal at line 1, column 6
valheim_1  | /home/steam/scripts/auto_backup.sh: line 19: [: : integer expression expected
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220303-011507-backup.tar.gz
valheim_1  | parse error: Invalid numeric literal at line 1, column 6
valheim_1  | /home/steam/scripts/auto_backup.sh: line 19: [: : integer expression expected
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220303-013006-backup.tar.gz
valheim_1  | parse error: Invalid numeric literal at line 1, column 6
valheim_1  | /home/steam/scripts/auto_backup.sh: line 19: [: : integer expression expected
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220303-014507-backup.tar.gz
valheim_1  | parse error: Invalid numeric literal at line 1, column 6
valheim_1  | /home/steam/scripts/auto_backup.sh: line 19: [: : integer expression expected
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220303-020006-backup.tar.gz
valheim_1  | parse error: Invalid numeric literal at line 1, column 6
valheim_1  | /home/steam/scripts/auto_backup.sh: line 19: [: : integer expression expected
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220303-021506-backup.tar.gz
valheim_1  | parse error: Invalid numeric literal at line 1, column 6
valheim_1  | /home/steam/scripts/auto_backup.sh: line 19: [: : integer expression expected
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220303-023006-backup.tar.gz
valheim_1  | parse error: Invalid numeric literal at line 1, column 6
valheim_1  | /home/steam/scripts/auto_backup.sh: line 19: [: : integer expression expected
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220303-024506-backup.tar.gz
valheim_1  | parse error: Invalid numeric literal at line 1, column 6
valheim_1  | /home/steam/scripts/auto_backup.sh: line 19: [: : integer expression expected
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220303-030006-backup.tar.gz
valheim_1  | parse error: Invalid numeric literal at line 1, column 6
valheim_1  | /home/steam/scripts/auto_backup.sh: line 19: [: : integer expression expected
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220303-031506-backup.tar.gz
valheim_1  | parse error: Invalid numeric literal at line 1, column 6
valheim_1  | /home/steam/scripts/auto_backup.sh: line 19: [: : integer expression expected
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220303-033006-backup.tar.gz
valheim_1  | parse error: Invalid numeric literal at line 1, column 6
valheim_1  | /home/steam/scripts/auto_backup.sh: line 19: [: : integer expression expected
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220303-034506-backup.tar.gz
valheim_1  | parse error: Invalid numeric literal at line 1, column 6
valheim_1  | /home/steam/scripts/auto_backup.sh: line 19: [: : integer expression expected
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220303-040006-backup.tar.gz
valheim_1  | parse error: Invalid numeric literal at line 1, column 6
valheim_1  | /home/steam/scripts/auto_backup.sh: line 19: [: : integer expression expected
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220303-041506-backup.tar.gz
valheim_1  | parse error: Invalid numeric literal at line 1, column 6
valheim_1  | /home/steam/scripts/auto_backup.sh: line 19: [: : integer expression expected
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220303-043007-backup.tar.gz
valheim_1  | parse error: Invalid numeric literal at line 1, column 6
valheim_1  | /home/steam/scripts/auto_backup.sh: line 19: [: : integer expression expected
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220303-044506-backup.tar.gz
valheim_1  | parse error: Invalid numeric literal at line 1, column 6
valheim_1  | /home/steam/scripts/auto_backup.sh: line 19: [: : integer expression expected
valheim_1  | [Valheim][steam]: Starting auto backup process...
valheim_1  | [Valheim][steam]: Removing old backups...
valheim_1  | [Valheim][steam]: Creating backup...
valheim_1  | [Valheim][steam]: Backup process complete! Created 20220303-050006-backup.tar.gz


</details>

**Reply by mbround18** [reply]
@lyk4ndna since you are on a rockpi you could try running with the configs @spannerman79 has goin on here: https://github.com/mbround18/valheim-docker/discussions/532#discussion-3889210

**Reply by lyk4ndna** [reply]
@mbround18 I looked at #532 and included the statements in the docker-compose file below. Are those the right spots to put them in? 


version: "3"
services:
valheim:
```yaml
     cap_add:
    - SYS_NICE
    cap_drop:
    - ALL
```
    image: mbround18/valheim:latest
    ports:
      - 2456:2456/udp
      - 2457:2457/udp
      - 2458:2458/udp
    environment:
      - PORT=2456
      - NAME=""
      - WORLD=""
      - PASSWORD=""
      - TYPE=ValheimPlus
      - "MODS=
       "
      - TZ=America/Chicago
      - PUBLIC=1
      - AUTO_UPDATE=1
      - AUTO_UPDATE_SCHEDULE="5 8 * * *"
      - AUTO_BACKUP=1
      - AUTO_BACKUP_SCHEDULE="*/15 * * * *"
      - AUTO_BACKUP_DAYS_TO_LIVE=1
      - AUTO_BACKUP_PAUSE_WITH_NO_PLAYERS=1
      - WEBHOOK_URL=""
    volumes:
    - ./valheim/saves:/home/steam/.config/unity3d/IronGate/Valheim
    - ./valheim/server:/home/steam/valheim
    - ./valheim/backups:/home/steam/backups

**Reply by spannerman79** [reply]
```yaml
cap_drop:
  - ALL
```

This part isn't required for docker-compose.  Only if you use `docker run` and use `--privileged` within that docker run.  As I **clearly** stated.

@lyk4ndna learn how to use markdown.  it helps break up that block of text that you are showing - a block of text in a partuclar format - in this case yaml.  See above for what I am talking about.

-----

But this is all academic unless the backup process is started with `nice` defined, and there is an env var able to be defined or if the nice value is already set to `19`.

**Reply by lyk4ndna** [reply]
@spannerman79 Edited the previous post, hope that's what you meant by breaking it up with markdown.

**Reply by spannerman79** [reply]
@lyk4ndna there is currently PR https://github.com/mbround18/valheim-docker/pull/549 that once merged will allow you to set the nice value to the auto backup process.

----

For others that may read this, if you do not wish to use this then it won't affect any pre-existing backup process you have in place (rsync/syncthing, etc).

**Reply by spannerman79** [reply]
@lyk4ndna the PR has been merged. Update/pull the latest image.

Add the following to your docker-compose.yml

```yaml
cap_add:
    - SYS_NICE
environment:
    - AUTO_BACKUP_NICE_LEVEL=19
```

**Reply by lyk4ndna** [reply]
@spannerman79  Awesome, thank you! Just pulled it and will see how it goes.

**Reply by spannerman79** [reply]
How did it go @lyk4ndna ? Its been a little over a week since your last reply here.
